content/articles/2019-01-tibble-2-0-0.Rmarkdown

```diff
@@ -0,0 +1,187 @@
+  - a function: apply custom name repair (e.g., `.name_repair = make.names` or `.name_repair = ~make.names(., unique = TRUE)` for names in the style of base R).
+
+```{r repair-names}
+## by default, duplicate names are not allowed
+tibble(` ` = 1, ` ` = 2)
+## you can authorize duplicate names
+tibble(` ` = 1, ` ` = 2, .name_repair = "minimal")
+## or request that the names be made unique
+tibble(` ` = 1, ` ` = 2, .name_repair = "unique")
+## or or universal
+tibble(` ` = 1, ` ` = 2, .name_repair = "universal")
+```
+
+Row name handling is stricter.
+Row names are never (and never were) supported in `tibble()` and `new_tibble()`, and are now stripped by default in `as_tibble()`.
+The `rownames` argument to `as_tibble()` supports:
+
+  - `NULL`: remove row names (default),
+  - `NA`: keep row names,
+  - A string: the name of the new column that will contain the existing row names, which are no longer present in the result.
+    
+  The old default can be restored by calling `pkgconfig::set_config("tibble::rownames", NA)`, this also works for packages that import _tibble_.
+    
+```{r row-names}
+rownames(as_tibble(mtcars))
+as_tibble(mtcars, rownames = "make_model")
+```
+
+The `glimpse()` function shows information obtained from `tbl_sum()` in the header, e.g. grouping information for `grouped_df` from _dplyr_, or other information from packages that override the `tbl_df` class.
+
+```{r}
+iris %>%
+  dplyr::group_by(Species) %>%
+  glimpse()
+```
+
+
+## Outlook
+
+The plan is to [use _vctrs_](https://github.com/tidyverse/tibble/issues/521) in _tibble_ 2.1.0.
+This package is a solid foundation for handling coercion, concatenation and recycling in vectors of arbitrary type.
+The support provided by _vctrs_ will yield a better `add_row()` implementation, in return name repair which is currently defined in _tibble_ should likely live in _vctrs_.
+
+Currently, installing _tibble_ can bring in almost dozen other packages:
+
+```{r deps}
+tools::package_dependencies("tibble", recursive = TRUE, which = "Imports")
+```
+
+Some of them, namely _fansi_ and _utf8_, are fairly heavy.
+[The plan](https://github.com/tidyverse/tibble/issues/475) is to make these packages, and _crayon_, a suggested package to _cli_, and provide fallback implementations there.
+When finished, taking a strong dependency on _tibble_ won't add too many new dependencies (again): _rlang_, _vctrs_ and _cli_ will be used by most of the tidyverse anyway, _pillar_ is the only truly new strong dependency.
+Packages that subclass `tbl_df` should import _tibble_ to make sure that the subsetting operator `[` always behaves the same.
+Constructing (subclasses of) tibbles should happen through `new_tibble()` only.
+
+Tibbles have a very opinionated way to print their data, not always in line with users' expectations, and sometimes clearly wrong (e.g. for numerical data where the absolute mean is much larger than the standard deviation).
```

Comment:
I think it's probably worth mentioning there's a connection to vctrs here, as vctrs makes constructing new S3 vector classes much easier.